frontend/src/modules/frameworkImport/views/frameworkImportView.js

```diff
@@ -75,23 +75,24 @@ define(function(require){
       Origin.router.navigateToHome();
     },
 
-    onFormSubmitSuccess: function(importData, impoprtStatus, importXhr) {
-      Origin.router.navigateToHome();
-    },
-
-    onAjaxSuccess: function() {
-      this.goBack();
+    onFormSubmitSuccess: function(data, impoprtStatus, importXhr) {
+      var title = data.title || '';
+      var msg = data.body.replace(/\n/g, "<br />") || '';
+      this.promptUser(title, msg, false);
     },
 
     onAjaxError: function(data, status, error) {
+      var title = data.responseJSON.title || Origin.l10n.t('app.importerrortitle');
+      var msg = data.responseJSON.body.replace(/\n/g, "<br />") || error;
+      this.promptUser(title, msg, true);
+    },
+    
+    promptUser: function(title, message, isError) {
       Origin.trigger('sidebar:resetButtons');
-      var text = (data.responseText) ? data.responseText.replace(/\n/g, "<br />") : error;
-
-      Origin.Notify.alert({
```

Comment:
Can we keep an alert, but just summarise the error, and instruct the user to view the full details in the `.server-log`.